ingest/src/main/scala/hydra.ingest/programs/TopicDeletionProgram.scala

```diff
@@ -63,39 +65,121 @@ final class TopicDeletionProgram[F[_]: MonadError[*[_], Throwable]](kafkaAdmin:
       }
     }
   }
+  //consumer group should be pulled from prod, consumer group name will work for what you want it to work for
+
+  def hasTopicReceivedRecentRecords(topicName: String): F[(String, List[Boolean])] = {
+    val curTime = Instant.now().toEpochMilli
+    kafkaAdmin.getLatestOffsets(topicName).flatMap {
+      offsetsMap => {
+        offsetsMap.toList.filterNot {
+          // remove partitions where offset == 0
+          _._2.value == 0
+        }.traverse {
+          case (partition, offset) => {
+            val tp = new TopicPartition(partition.topic, partition.partition)
+            // get latest published message, compare timestamp to (NOW - configurable time window)
+            kafkaClient.streamStringKeyFromGivenPartitionAndOffset(topicName, "dvs.deletion.consumer.group", false,
+              List((tp, offset.value - 1))).take(1).map { case (_, _, timestamp) =>
+                timestamp.createTime.getOrElse(0: Long) < curTime - allowableTopicDeletionTime
+            }.compile.lastOrError
+          }
+          case _ =>
+            throw new Throwable("Unexpected return type from KafkaAdminAlgebra#getLatestOffsets")
+        }
+      }
+    }.map {list => topicName -> list}
+  }
 
-  def deleteTopic(topicNames: List[String], ignoreConsumerGroups: List[String]): F[ValidatedNel[DeleteTopicError, Unit]] = {
-    val eitherErrorOrTopic = checkIfTopicStillHasConsumers(topicNames, ignoreConsumerGroups)
-    eitherErrorOrTopic.flatMap{ cge =>
-      val goodTopics: List[String] = cge.map { e =>
-        e match {
-          case Right(value) => value
-          case Left(_) => ""
+  def checkForActivelyPublishedTopics(topicNames: List[String]): F[List[Either[ActivelyPublishedToError, String]]] = {
+    topicNames.traverse { name =>
+      val topicsResultsF = hasTopicReceivedRecentRecords(name)
+
+      for {
+        topicResults <- topicsResultsF
+      } yield {
+        if(topicResults._2.contains(false)){
+          Left(ActivelyPublishedToError(topicResults._1, allowableTopicDeletionTime))
+        } else {
+          Right(topicResults._1)
         }
+      }
+    }
+  }
+
+  def findDeletableTopics(eitherErrorOrTopicAPTE: List[Either[ActivelyPublishedToError, String]],
+                          eitherErrorOrTopicCSEE: List[Either[ConsumersStillExistError, String]]): List[String] = {
+    def getOnlyTopics(list: List[Either[DeleteTopicError, String]]): List[String] = {
+      list.map {
```

Comment:
In Scala, if you are ever filtering something out, don't use String types. Use option. In this case you can do `list.map(_.toOption).flatten` instead of lines 112 - 116.

Reply:
For context: calling toOption on an Either returns Some(Right) or None. So in your case Some(String) or None. It turns the error (left side) into None. Then the flatten removes all the option stuff resulting in just the Strings, none of the errors.

Reply:
Implemented changes as described.